tests/fe/fe_enriched_step-36.cc

```diff
@@ -0,0 +1,948 @@
+              else if (cell->neighbor(f)->has_children() == true)
+                {
+                  // The neighbor does
+                  // have
+                  // children. See if
+                  // any of the cells
+                  // on the other
+                  // side are vanilla FEM
+                  for (unsigned int sf=0; sf<cell->face(f)->n_children(); ++sf)
+                    if (!cell_is_pou (cell->neighbor_child_on_subface(f, sf)))
+                      {
+                        face_is_on_interface = true;
+                        break;
+                      }
+                }
+              // add constraints
+              if (face_is_on_interface)
+                {
+                  cell->face(f)->get_dof_indices (local_face_dof_indices, pou_fe_index);
+                  for (unsigned int i=0; i<local_face_dof_indices.size(); ++i)
+                    if (fe_collection[pou_fe_index].face_system_to_base_index(i).first.first == pou_group)
+                      //if (fe_collection[1].face_system_to_component_index(i).first /*component*/ > 0)
+                      constraints.add_line (local_face_dof_indices[i]);
+                }
+            }
+  }
+
+  template <int dim>
+  void
+  EigenvalueProblem<dim>::assemble_system()
+  {
+    stiffness_matrix = 0;
+    mass_matrix = 0;
+
+    dealii::FullMatrix<double> cell_stiffness_matrix;
+    dealii::FullMatrix<double> cell_mass_matrix;
+    std::vector<types::global_dof_index> local_dof_indices;
+    std::vector<double> potential_values;
+    std::vector<double> enrichment_values;
+    std::vector<Tensor<1,dim>> enrichment_gradients;
+
+    hp::FEValues<dim> fe_values_hp(fe_collection, q_collection,
+                                   update_values | update_gradients |
+                                   update_quadrature_points | update_JxW_values);
+
+
+    typename hp::DoFHandler<dim>::active_cell_iterator
+    cell = dof_handler.begin_active (),
+    endc = dof_handler.end ();
+    for (; cell!=endc; ++cell)
+      if (cell->subdomain_id() == this_mpi_process )
+        {
+          fe_values_hp.reinit (cell);
+          const FEValues<dim> &fe_values = fe_values_hp.get_present_fe_values();
+
+          const unsigned int &dofs_per_cell = cell->get_fe().dofs_per_cell;
+          const unsigned int &n_q_points    = fe_values.n_quadrature_points;
+
+          potential_values.resize(n_q_points);
+          enrichment_values.resize(n_q_points);
+          enrichment_gradients.resize(n_q_points);
+
+          local_dof_indices.resize     (dofs_per_cell);
+          cell_stiffness_matrix.reinit (dofs_per_cell,dofs_per_cell);
+          cell_mass_matrix.reinit      (dofs_per_cell,dofs_per_cell);
+
+          cell_stiffness_matrix = 0;
+          cell_mass_matrix      = 0;
+
+          potential.value_list (fe_values.get_quadrature_points(),
+                                potential_values);
+
+#ifdef WITHOUT_FE_ENRICHED
```

Comment:
I think it would be useful to figure out a way how we can run this test with both variants -- because that would give us one more test. One way would be to clone it (including code duplication) and in each of the two `.cc` files delete one branch of the `#ifdef`, once you're satisfied that you've got the final version of the test. There are of course also other ways to avoid the code duplication.